cmd/wharf/aggregator.go

```diff
@@ -20,6 +20,8 @@ var aggregatorCmd = &cobra.Command{
 After streaming from a worker is done, the aggregator will kill it using the
 kill endpoint.`,
 	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
+		go handleCancelSignals(rootCancel)
+
```

Comment:
Suggest moving this into a PersistentPreRun on the `rootCmd` instead, to reduce the duplication.

Reply:
Aggregator seems to require it because only the closest defined `PersistentPreRun` is executed, so for `wharf aggregator serve` only the one defined on `wharf aggregator` gets executed.

Same with provisioner.

Reply:
I hereby name that as "stupid".

It's persistent :angry: 

We could call it manually though, which I suggest we do instead. Something like this:

```go
var sub2Cmd = &cobra.Command{
	Use: "sub2",
	Run: func(cmd *cobra.Command, args []string) {
		fmt.Println("sub2 called")
	},
	PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
		if err := callParentPersistentPreRuns(cmd, args); err != nil {
			return err
		}
		fmt.Println("sub2 PersistentPreRun")
	},
	TraverseChildren: true,
}

func callParentPersistentPreRuns(cmd *cobra.Command, args []string) error {
	cmd = cmd.Parent()
	if cmd == nil {
		return nil
	}
	// Call the root-most one first
	if err := callParentPersistentPreRuns(cmd, args); err != nil {
		return err
	}
	if cmd.PersistentPreRun != nil {
		cmd.PersistentPreRun(cmd, args)
	}
	if cmd.PersistentPreRunE != nil {
		return cmd.PersistentPreRunE(cmd, args)
	}
	return nil
}
```

Not that pretty, but I think it'll keep things easier to organize

Reply:
Yeah, I was pretty confuddled by it for a bit.
Feels more like it should be an opt-out.

Will implement your suggested change 👍🏻